internal/services/signalr/web_pubsub_custom_certificate_binding.go

```diff
@@ -0,0 +1,282 @@
+				Target:     []string{string(webpubsub.ProvisioningStateSucceeded)},
+				Refresh:    webPubsubCustomCertBindingStateRefreshFunc(ctx, client, id),
+				MinTimeout: 15 * time.Second,
+				Timeout:    30 * time.Minute,
+			}
+
+			if _, err := stateConf.WaitForStateContext(ctx); err != nil {
+				return fmt.Errorf("waiting for creation of %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r CustomCertBindingWebPubsubResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.SignalR.WebPubSubClient.WebPubSub
+			keyVaultClient := metadata.Client.KeyVault
+			resourcesClient := metadata.Client.Resource
+			id, err := webpubsub.ParseCustomCertificateID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.CustomCertificatesGet(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading web pubsub custom certificate binding %s: %+v", id, err)
+			}
+
+			if resp.Model == nil {
+				return fmt.Errorf("retrieving %s: got nil model", *id)
+			}
+
+			vaultBasedUri := resp.Model.Properties.KeyVaultBaseUri
+			certName := resp.Model.Properties.KeyVaultSecretName
+
+			keyVaultIdRaw, err := keyVaultClient.KeyVaultIDFromBaseUrl(ctx, resourcesClient, vaultBasedUri)
+			if err != nil {
+				return fmt.Errorf("getting key vault base uri from %s: %+v", id, err)
+			}
+			vaultId, err := keyVaultParse.VaultID(*keyVaultIdRaw)
+			if err != nil {
+				return fmt.Errorf("parsing key vault %s: %+v", vaultId, err)
+			}
+
+			webPubsubId := webpubsub.NewWebPubSubID(id.SubscriptionId, id.ResourceGroupName, id.WebPubSubName).ID()
+
+			certVersion := ""
+			if resp.Model.Properties.KeyVaultSecretVersion != nil {
+				certVersion = *resp.Model.Properties.KeyVaultSecretVersion
+			}
+			nestedItem, err := keyVaultParse.NewNestedItemID(vaultBasedUri, "certificates", certName, certVersion)
+			if err != nil {
+				return err
+			}
+
+			certId := nestedItem.ID()
+
+			state := CustomCertBindingWebPubsubModel{
+				Name:               id.CustomCertificateName,
+				CustomCertId:       certId,
+				WebPubsubId:        webPubsubId,
+				CertificateVersion: utils.NormalizeNilableString(resp.Model.Properties.KeyVaultSecretVersion),
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r CustomCertBindingWebPubsubResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.SignalR.WebPubSubClient.WebPubSub
+
+			id, err := webpubsub.ParseCustomCertificateID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var webPubsubCustomCertBinding CustomCertBindingWebPubsubModel
```

Comment:
Suggest keep consistent with other resources
```suggestion
			var model CustomCertBindingWebPubsubModel
```